src/Components/Web.JS/src/Platform/Circuits/BlazorOptions.ts

```diff
@@ -0,0 +1,40 @@
+import { LogLevel } from '../Logging/Logger';
+
+export interface BlazorOptions {
+  configureSignalR: (builder: signalR.HubConnectionBuilder) => void;
+  logLevel: LogLevel;
+  reconnectionOptions: ReconnectionOptions;
+  reconnectionHandler?: ReconnectionHandler;
+}
+
+export function resolveOptions(userOptions?: Partial<BlazorOptions>): BlazorOptions {
+    const result = { ...defaultOptions, ...userOptions };
+
+    // The spread operator can't be used for a deep merge, so do the same for subproperties
+    if (userOptions && userOptions.reconnectionOptions) {
+      result.reconnectionOptions = { ...defaultOptions.reconnectionOptions, ...userOptions.reconnectionOptions };
+    }
+
+    return result;
+}
+
+export interface ReconnectionOptions {
+  maxRetries: number;
+  retryIntervalMilliseconds: number;
+  dialogId: string;
+}
+
+export interface ReconnectionHandler {
+  onConnectionDown(options: ReconnectionOptions, error?: Error): void;
+  onConnectionUp(): void;
+}
+
+const defaultOptions: BlazorOptions = {
+    configureSignalR: (_) => { },
```

Comment:
Nit, formtting seems to be off here (4 spaces versus 2)?

Reply:
Similarly in `resolveOptions`

Reply:
Good point but would rather not run the CI again over this. We have more work to do in this area soon and can handle this then.